stock_picking_send_by_mail/models/stock_picking.py

```diff
@@ -8,21 +8,39 @@
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
-    def action_picking_send(self):
+    def _get_picking_send_email_template(self):
+        return self.company_id.stock_mail_confirmation_template_id
+
+    def action_picking_send(self, send=False):
+        """Send an e-mail from a delivery transfer.
+
+        :param send: by default this method is opening the email composer wizard.
+            If set to `True`, the email will be sent directly without confirmation.
+        """
         self.ensure_one()
-        template = self.company_id.stock_mail_confirmation_template_id
+        template = self._get_picking_send_email_template()
         compose_form = self.env.ref(
             "mail.email_compose_message_wizard_form",
             False,
         )
+        template_id = template.id if template else False
         ctx = dict(
             default_model="stock.picking",
             default_res_id=self.id,
             default_use_template=bool(template),
-            default_template_id=template and template.id or False,
+            default_template_id=template_id,
             default_composition_mode="comment",
             user_id=self.env.user.id,
         )
+        if send:
```

Comment:
I'd move this to its own method and I would enable this w/ a flag on the picking type

Reply:
forgot this part!